### ISSUE TITLE
Support merging LoRA/AdaLoRA weights into the base model.

### ISSUE THREAD

**Issue by arnavgarg1** [ludwig-ai/ludwig#3603]
Once the model is fine-tuned using an adapter, it would be great for Ludwig to natively support merging the LoRA weights into the base model directly instead of only supporting the option of keeping the adapter weights separate. 

Ideally, we want to support both the option to:

1. Save adapter weights independent of the base model (good for folks who use a base model with many different fine-tuned adapters)
2. Merge adapter weights into the base model, then save the merged base model (good if you want to be able to pull your model easily from HF for inference or even use HF inference endpoints).

Implementation-wise, this is a pretty simple change.

We want to add a new function to the [LLM base model class](https://github.com/ludwig-ai/ludwig/blob/master/ludwig/models/llm.py#L106) called `merge_and_unload` that:

1. Checks if the model is loaded in with an adapter
2. Checks if the adapter class is LoRA or AdaLoRA
3. Calls model.merge_and_unload()

Pseudocode might look something like this once it is implemented:
```python
model = LudwigModel.load("</path/to/trained/model>")
merged_model = model.model.merge_and_unload()
merged_model.save_pretrained("<path_to_save_on_disk>")
```

**Comment by alexsherstinsky** [comment]
#self-assign

**Comment by alexsherstinsky** [comment]
#take

**Comment by alexsherstinsky** [comment]
This has been implemented in #3649 -- thank you.
